File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        result=0
        openbracket=0
        for char in s:
            if char=='(':
                openbracket +=1
                result=max(result,openbracket)
            elif char==')':
                openbracket -=1
        return result
        

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int opencount=0;
        int ans=0;
        for(char ch:s.toCharArray()){
            if(ch=='('){
                opencount++;
            }
            else if(ch==')'){
                opencount--;
            }
            ans=Math.max(ans,opencount);
        }
        return ans;
    }
}